### PR TITLE
feat: icss import init

### PR DESCRIPTION
**Pull request by faga295**
**What kind of change does this PR introduce?**
support `:import` block

**Did you add tests for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

### CODE REVIEW

File: lib/css/CssParser.js [hunk 1]
```diff
@@ -368,6 +371,74 @@ class CssParser extends Parser {
...
+					);
+					return pos;
+				}
+				data.imports[name] = { value, replaceRanges: [] };
+			}
+			pos++;
+			data.range = [start, pos];
+			icssImportData.push(data);
+			return pos;
+		};
```
Comment (alexander-akait) [inline lib/css/CssParser.js:441]: Avoid such things, you can use set `let inIcssImportSelector` when you found `:import`, and collect all things, you can use more `is*` helpers, avoid any changes in tokenizer logic

Comment (faga295) [reply lib/css/CssParser.js:441]: Thanks for your suggestion, these code is inspired by `parseExports`, maybe it needs to be changed as well.

File: lib/css/CssParser.js [hunk 2]
```diff
@@ -510,12 +518,10 @@ class CssParser extends Parser {
 
 							break;
 						}
-
 						importData.url = normalizeUrl(
 							input.slice(start + 1, end - 1),
 							true
 						);
-
```
Comment (alexander-akait) [inline lib/css/CssParser.js:518]: Avoid removing these changes, like removing lines/saces/etc

File: lib/css/CssParser.js [hunk 3]
```diff
@@ -815,12 +859,29 @@ class CssParser extends Parser {
...
 				}
 				return end;
 			},
 			identifier: (input, start, end) => {
 				switch (scope) {
 					case CSS_MODE_IN_BLOCK: {
+						if (icssImportsData) {
+							const dataIndex = icssImportsData.findIndex(data =>
+								data.importMap.get(input.slice(start, end))
+							);
```
Comment (alexander-akait) [inline lib/css/CssParser.js:878]: Never use `findIndex`, if you want to use it means you do something wrong, wrong place when you replace (too later)

File: lib/css/CssParser.js [hunk 4]
```diff
@@ -1040,7 +1111,19 @@ class CssParser extends Parser {
 		if (!this.namedExports) {
 			module.buildMeta.defaultObject = "redirect";
 		}
-
+		if (icssImportsData.length) {
+			for (const data of icssImportsData) {
```
Comment (alexander-akait) [inline lib/css/CssParser.js:1115]: please create deps where you need and avoid any loops, it is very slow

File: lib/dependencies/CssIcssImportDependency.js [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	 * @param {ImportMap} importMap collected import data of dependency
+	 */
+	constructor(request, range, importMap) {
+		super(request);
+		this.range = range;
+		this.importMap = importMap;
+	}
+
+	get type() {
+		return "css :import";
```
Comment (alexander-akait) [inline lib/dependencies/CssIcssImportDependency.js:36]: Use `css-:import`

File: lib/css/CssModulesPlugin.js [hunk 6]
```diff
@@ -826,25 +823,108 @@ class CssModulesPlugin {
...
+						? Array.from(
+								exports,
+								([n, v]) => `${escapeCss(n)}:${escapeCss(v)}/`
+							).join("")
+						: ""
+				}${esModule ? "&" : ""}${escapeCss(moduleId)}`
+			);
+
+			source.add(replaceSource);
+		}
```
Comment (alexander-akait) [inline lib/css/CssModulesPlugin.js:926]: It is very slow, we need to search another approach

Comment (faga295) [reply lib/css/CssModulesPlugin.js:926]: Maybe we can define  a CSS variable for each CSS export, and each ICSS import will be replaced by the CSS variable

Comment (alexander-akait) [reply lib/css/CssModulesPlugin.js:926]: except for cases when they are really needed

Comment (faga295) [reply lib/css/CssModulesPlugin.js:926]: I will try again to collect replacementRange and then replace them during renderChunk. However, this requires ensuring that the source remains as replaceSource from the generation phase before applying replacements. Currently, `InitFragment.addToSource` in generate and renderModule both modify the source, so I plan to move these operations after replacing the range.

**Pull request closed without merging**